### PR TITLE
fix: welcome form being submitted on pressing enter

### PR DESCRIPTION
**Pull request by akash-codemonk**
## Description

`<form>`'s get submitted when pressed on enter in any of the inputs. With this change, if validation passes only then onSubmit is called.

Fixes https://github.com/appsmithorg/appsmith/issues/22184

### Deploy preview
https://ce-22338.dp.appsmith.com/

## Type of change

- Bug fix (non-breaking change which fixes an issue

## How Has This Been Tested?
- Manual

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

**Comment by Richarex** [comment]
This issue is fixed now, verified it on the DP url.

File: app/client/src/pages/setup/GetStarted.tsx [hunk 1]
```diff
@@ -84,8 +84,16 @@ const DROPDOWN_WIDTH = "400px";
...
+    props.onGetStarted &&
+      props.onGetStarted(
+        data.role !== "other" ? data.role : props.role_name,
+        data.useCase,
+      );
+  };
+
   return (
-    <StyledNonSuperUserForm>
+    <StyledNonSuperUserForm onSubmit={props.handleSubmit(onSubmit)}>
```
Comment (akash-codemonk) [inline app/client/src/pages/setup/GetStarted.tsx:96]: onSubmit is called only if it is valid here.

File: app/client/src/pages/setup/GetStarted.tsx [hunk 2]
```diff
@@ -118,15 +126,9 @@ function NonSuperUser(
...
           disabled={props.invalid}
-          onClick={() =>
-            !props.invalid && // temp fix - design system needs to be fixed for disabling click
-            props.onGetStarted &&
-            props.onGetStarted(
-              props.role !== "other" ? props.role : props.role_name,
-              props.useCase,
-            )
-          }
+          tag="button"
```
Comment (akash-codemonk) [inline app/client/src/pages/setup/GetStarted.tsx:129]: Changed to button(by default ds button is an anchor tag) and added type submit to trigger onsubmit on click.

File: app/client/src/pages/setup/GetStarted.tsx [hunk 3]
```diff
@@ -84,8 +84,16 @@ const DROPDOWN_WIDTH = "400px";
 function NonSuperUser(
   props: InjectedFormProps & UserFormProps & NonSuperUserFormData,
 ) {
+  const onSubmit = (data: NonSuperUserFormData) => {
```
Comment (akash-codemonk) [inline app/client/src/pages/setup/GetStarted.tsx:87]: onSubmit is called only if it is valid here.

File: app/client/src/pages/setup/GetStarted.tsx (hunk 2 above)
Comment (hetunandu) [reply app/client/src/pages/setup/GetStarted.tsx:129]: Can you confirm if this is using the the new DS button?

Comment (akash-codemonk) [reply app/client/src/pages/setup/GetStarted.tsx:129]: Yup it has the new ds version with the fix for the issue where `onClick` was being called even when button was disabled.

**Review by hetunandu — APPROVED** [review]

**Pull request merged**